sttp/client/src/main/scala/endpoints4s/sttp/client/Endpoints.scala

```diff
@@ -332,11 +323,15 @@ trait EndpointsWithCustomErrors[R[_]]
       //#endpoint-type
       {
     def apply(a: A): R[B] = {
-      val result = backend.send(request(a).response(asStringAlways))
-      backend.responseMonad.flatMap(result) { sttpResponse =>
-        decodeResponse(response, sttpResponse)
+        val result = backend.send(request(a).response(asStringAlways))
+        val responseFuture = backend.responseMonad.handleError(result) {
+          case e if e != null && e.getCause.getClass.getName.toLowerCase.contains("timeout") =>
```

Comment:
Would it be possible to pattern match on the cause using a “typed pattern” instead of manipulating the class name? Something like:

~~~ scala
e.getCause match {
  case timeout: TimeoutException => ...
}
~~~